Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -245,39 +257,47 @@ private void OnGameChange(GameRunLevelChangedEvent ev)
 
     #region Private API
 
-    private void SendEntitySpeak(EntityUid source, string originalMessage, bool hideChat = false)
+    private void SendEntitySpeak(EntityUid source, string originalMessage, bool hideChat, bool hideGlobalGhostChat, string? nameOverride)
     {
         if (!_actionBlocker.CanSpeak(source))
             return;
 
         var (message, channel) = GetRadioPrefix(source, originalMessage);
 
+        message = TransformSpeech(source, message);
+        if (message.Length == 0)
+            return;
+
         if (channel != null)
         {
-            _listener.PingListeners(source, message, channel);
-            SendEntityWhisper(source, message, hideChat);
+            SendEntityWhisper(source, message, hideChat, hideGlobalGhostChat, channel, nameOverride);
```

Comment:
Would this get double transformed? As whisper is being passed the modified message.

Reply:
yes it would, fixed now.